### PR TITLE
Solve the jobid is empty cause canceled inaccurate

### PR DESCRIPTION
**Pull request by MonsterChenzhuo**
Problem Summary:

### What is changed and how it works?

Proposal: [xxx](url) 

Feel free to ping committers for the review!

## Contribution Checklist

  - Make sure that the pull request corresponds to a [GITHUB issue](https://github.com/apache/streampark/issues).

  - Name the pull request in the form "[Feature] [component] Title of the pull request", where *Feature* can be replaced by `Hotfix`, `Bug`, etc.

  - Minor fixes should be named following this pattern: `[hotfix] [docs] Fix typo in README.md doc`.

### CODE REVIEW

File: streampark-flink/streampark-flink-kubernetes/src/main/scala/org/apache/streampark/flink/kubernetes/watcher/FlinkJobStatusWatcher.scala [hunk 1]
```diff
@@ -234,12 +235,15 @@ class FlinkJobStatusWatcher(conf: JobStatusWatcherConfig = JobStatusWatcherConfi
    * list flink jobs details from rest api
    */
   private def callJobsOverviewsApi(restUrl: String): Option[JobDetails] = {
-    JobDetails.as(
+    logger.info(s"Try to access flink's service via http:${restUrl}/jobs/overview.")
```
Comment (xinzhuxiansheng) [inline streampark-flink/streampark-flink-kubernetes/src/main/scala/org/apache/streampark/flink/kubernetes/watcher/FlinkJobStatusWatcher.scala]: Is it better to change the log here to debug level?  Job status monitoring is continuous, reducing unnecessary logs as much as possible

**Comment by xinzhuxiansheng** [comment]
@MonsterChenzhuo  
The job status monitoring is polling. Under normal circumstances, the info inside will always be printed. Can some debug logs be changed to debug level?

作业状态监听是轮询的，对于正常情况下，里面的info会一直打印，可否将一些调试的log，改成debug级别

**Comment by MonsterChenzhuo** [comment]
> @MonsterChenzhuo The job status monitoring is polling. Under normal circumstances, the info inside will always be printed. Can some debug logs be changed to debug level?
> 
> 作业状态监听是轮询的，对于正常情况下，里面的info会一直打印，可否将一些调试的log，改成debug级别

ok👌

**Comment by xinzhuxiansheng** [comment]
@MonsterChenzhuo      In the K8sFlinkChangeEventListener.updateApplicationWithJobStatusCV()  method, is it possible to consider not to assign a value to the jobId of the app? 

在K8sFlinkChangeEventListener.updateApplicationWithJobStatusCV()  方法中，是否可以考虑不用再给app的jobId赋值了

**Comment by MonsterChenzhuo** [comment]
> updateApplicationWithJobStatusCV
A global search found two places to setJobId, which have been modified
全局搜索了一下 发现有两个地方都有去setJobId，已经进行了修改

**Review by xinzhuxiansheng — COMMENTED** [review]
i have finished review : )

File: streampark-flink/streampark-flink-submit/streampark-flink-submit-core/src/main/scala/org/apache/streampark/flink/submit/trait/FlinkSubmitTrait.scala [hunk 2]
```diff
@@ -98,7 +98,7 @@ trait FlinkSubmitTrait extends Logger {
       .safeSet(CoreOptions.CLASSLOADER_RESOLVE_ORDER, submitRequest.resolveOrder.getName)
       .safeSet(ApplicationConfiguration.APPLICATION_MAIN_CLASS, submitRequest.appMain)
       .safeSet(ApplicationConfiguration.APPLICATION_ARGS, extractProgramArgs(submitRequest))
-      .safeSet(PipelineOptionsInternal.PIPELINE_FIXED_JOB_ID, new JobID().toHexString)
+      .safeSet(PipelineOptionsInternal.PIPELINE_FIXED_JOB_ID, submitRequest.jobID)
```
Comment (xinzhuxiansheng) [inline streampark-flink/streampark-flink-submit/streampark-flink-submit-core/src/main/scala/org/apache/streampark/flink/submit/trait/FlinkSubmitTrait.scala]: submitRequest.jobID corresponds to application.getId() , you need to redefine a parameter flinkJobId, it is best to distinguish it from jobID, the naming here may be misleading。 

submitRequest.jobID 对应的是 application.getId() ,你需要在submitRequest 重新定义一个参数 flinkJobId, 最好能与jobID 区分开，这里的命名可能会产生误导

refer:
ApplicationServiceImpl.java
```
        Map<String, String> dynamicOption = FlinkSubmitter.extractDynamicOptionAsJava(application.getDynamicOptions());

        Map<String, Object> extraParameter = new HashMap<>(0);
        extraParameter.put(ConfigConst.KEY_JOB_ID(), application.getId());

        if (appParam.getAllowNonRestored()) {
            extraParameter.put(SavepointConfigOptions.SAVEPOINT_IGNORE_UNCLAIMED_STATE.key(), true);
        }
```

**Comment by lvshaokang** [comment]
@MonsterChenzhuo LGTM. Thanks.

**Review by wolfboys — APPROVED** [review]
LGTM

**Pull request merged**